### PR TITLE
Bump React to 16.7.0

### PR DESCRIPTION
**Pull request by mwiencek**
Just trying to not lag too far behind before it becomes a pain to update.

There should be no breaking changes for us. There are a ton of bug fixes, a few new features, and a couple fixed vulnerabilities.

### CODE REVIEW

**Comment by reosarevok** [comment]
What does the new scheduler do? (since it's not mentioned in the desc)

**Comment by mwiencek** [comment]
> What does the new scheduler do? (since it's not mentioned in the desc)

Not entirely sure, but it has something to do with https://reactjs.org/blog/2018/11/27/react-16-roadmap.html#react-16x-q2-2019-the-one-with-concurrent-mode which should be nice for improving responsiveness, though doesn't appear to be stable yet.

**Comment by mwiencek** [comment]
There was a bug fix to `getDerivedStateFromProps` in 16.4 which apparently broke a small number of peoples' code. I don't think we'd have been affected, but it turned out all our uses of this method were anti-patterns, anyway, so I removed all of them. See https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

File: root/components/common-macros.tt [hunk 1]
```diff
@@ -803,7 +803,7 @@ END ~%]
 -%]
     [%- React.embed(c, 'static/scripts/common/components/WikipediaExtract', {
         entity => {entityType => entity.entity_type, gid => entity.gid},
-        wikipediaExtract => wikipedia_extract,
+        cachedWikipediaExtract => wikipedia_extract,
```
Comment (mwiencek) [inline root/components/common-macros.tt]: This should help make clear what the prop's for, and that it won't affect the displayed extract if the prop changes. (The prop couldn't change here anyway.)

**Comment by mwiencek** [comment]
@brainzbot test this please

**Comment by mwiencek** [comment]
@brainzbot test this please

**Pull request merged**